done task_19

GeniusFromMinsk opened the pull request:
Добавил Model, DAO, Service, util, starter модули. Очень стыдно за starter, но не особо понимаю пока как его по другому написать(там просто тесты работы бд), функционал весь тестил, было без ошибок.

GeniusFromMinsk committed:
done task_19

Willynick commented:
Конфликты нужно порешать. Если есть проблемы с решением конфликтов - может созвониться
![image](https://github.com/GeniusFromMinsk/taskmanager/assets/57830441/d7a89e06-0d35-418f-8249-f26638d653bb)

Willynick commented on dao/pom.xml:
```diff
@@ -0,0 +1,59 @@
+    <packaging>jar</packaging>
+
+    <name>dao</name>
+    <url>http://maven.apache.org</url>
```
Убираем url отовсюду

Willynick commented on dao/.gitignore:
```diff
@@ -0,0 +1,38 @@
+target/
```
Скрипты перенеси в модуль dao. В resource папке там просто создай папку sql и туда скрипты
![image](https://github.com/GeniusFromMinsk/taskmanager/assets/57830441/bb555b91-847c-4d60-8aa7-08a4fea5fe6a)

Willynick replied:
А src каталог вообще удалить из главного модуля

Willynick commented on dao/.gitignore:
```diff
@@ -0,0 +1,38 @@
+*.iml
+*.ipr
+
+### Eclipse ###
```
.gitignore файл оставляй только в родительском модуле
![image](https://github.com/GeniusFromMinsk/taskmanager/assets/57830441/cd6c9b6a-0c88-420e-8655-f4f5f3b61c04)

Willynick commented on dao/src/test/java/com/itacademy/courses/AppTest.java:
```diff
@@ -0,0 +1,38 @@
+/**
+ * Unit test for simple App.
+ */
+public class AppTest 
```
Убираем везде сгенерированные тесты
![image](https://github.com/GeniusFromMinsk/taskmanager/assets/57830441/5b4793c5-9b48-47e5-9d38-62779ee7f746)

Willynick commented on util/src/main/java/com/itacademy/courses/db/DBConnection.java:
```diff
@@ -0,0 +1,15 @@
+import java.sql.SQLException;
+
+public class DBConnection {
+    private static final String JDBC_URL = "jdbc:postgresql://localhost:5432/manager";
```
Настроечки давай вынесем в отдельный файл

Willynick commented on util/src/main/java/com/itacademy/courses/db/DBConnection.java:
```diff
@@ -0,0 +1,15 @@
+    private static final String JDBC_PASSWORD = "1111";
+
+    public static Connection getConnection() throws SQLException {
+        return DriverManager.getConnection(JDBC_URL, JDBC_USER, JDBC_PASSWORD);
```
Каждый раз будешь создавать Connection?)
Попахивает потерей производительности
![image](https://github.com/GeniusFromMinsk/taskmanager/assets/57830441/05fd7630-9770-4553-9852-63fe8458e302)

Willynick replied:
Переделай на синглтон

Willynick commented on starter/src/main/java/com/itacademy/courses/Main.java:
```diff
@@ -0,0 +1,34 @@
+    public static void main(String[] args) throws SQLException, ParseException {
+
+        /*
+        InsertApp application = new InsertApp();
```
Почему всё закомментировано? Пусть останется, правда не stackTrace, а логгирование

Willynick commented on model/src/main/java/com/itacademy/courses/models/Category.java:
```diff
@@ -0,0 +1,49 @@
+    private String name;
+    private String description;
+
+    private Timestamp created_at;
```
Все в camelCase, в Java всегда и везде. Поменяй везде у себя

Willynick commented on model/src/main/java/com/itacademy/courses/models/Category.java:
```diff
@@ -0,0 +1,49 @@
+import java.sql.Timestamp;
+
+public class Category {
+    private int id;
```
Пустая строка после пятой строки. Применить так для всех классов

Willynick commented on model/src/main/java/com/itacademy/courses/models/Category.java:
```diff
@@ -0,0 +1,49 @@
+
+    public void setUpdatedAt(Timestamp updatedAt) {
+        this.updated_at = updatedAt;
+    }
```
Пустая строка после 48 строки. Применить так ко всем другим классам

Willynick commented on dao/src/main/java/com/itacademy/courses/dao/CategoryDAO.java:
```diff
@@ -0,0 +1,63 @@
+import java.util.List;
+
+public class CategoryDAO {
+    private static final String INSERT_CATEGORY_SQL = "INSERT INTO Categories (name, description) VALUES (?, ?)";
```
Пиши уже в скриптах всё в верхнем регистре, так будет красиво. Т.е. CATEGORIES, NAME, DESCRIPTION и т.д.

Willynick commented on dao/src/main/java/com/itacademy/courses/dao/CategoryDAO.java:
```diff
@@ -0,0 +1,63 @@
+        return rowUpdated;
+    }
+
+    private void printSQLException(SQLException ex) {
```
Создай утилитный класс, который будет отлавливать разные ошибки и писать сообщение об ошибке

Willynick commented on starter/src/main/java/com/itacademy/courses/InsertApp.java:
```diff
@@ -0,0 +1,177 @@
+    private void createFiler(){
+        Filter filter = new Filter();
+        filter.setName("Completed Tasks");
+        filter.setCriteria("status: completed");
```
Как ты потом будешь фильтровать по такой критерии? Продумай механизм, потому что фильтровать будешь ты. И по-хорошему тебе этот механизм нужно уже реализовать)

Willynick commented on service/src/main/java/com/itacademy/courses/services/UserService.java:
```diff
@@ -0,0 +1,41 @@
+import java.sql.SQLException;
+import java.util.List;
+
+public class UserService {
```
Тебе нужно уже хотя бы наметить будущие методы, а не просто вызывать методы ДАО. Т.е. метод регистрации пользователя, метод логина пользователя и весь остальной функционал, что предусмотрено твоей задачей. Т.е. сделать какую-то несложную логику первоначальную. Потом уже можем дорабатывать

GeniusFromMinsk committed:
исправленное task_19
Внёс изменения в структуру бд(удалил таблицу filters из ненадобности), потому что можно фильтровать из таблицы tasks по критериям)
Добавил логику User на регистрацию и авторизацию(если одинаковый логин или email, то регистрация не пройдет)
Также добавил exception директорию в Util где описал исключение.

Willynick commented on dao/src/main/java/com/itacademy/courses/dao/TaskDAO.java:
```diff
@@ -119,7 +120,7 @@ public boolean updateTask(Task task) throws SQLException {
     public List<Task> selectTasksByStatus(String status) {
         List<Task> tasks = new ArrayList<>();
-        try (Connection connection = DBConnection.getConnection();
+        try (Connection connection = DBConnection.getInstance().getConnection();
```
У тебя может быть много фильтров. Давай их объединим в enum, где каждой константе енама будет соответствовать нужны SQL (часть SQL) запроса. Создай сервисный метод, который будет принимать значение енама и по нему филтровать

GeniusFromMinsk replied:
Гениально хавхавхахва, реально лайк

Willynick commented on service/src/main/java/com/itacademy/courses/services/UserService.java:
```diff
@@ -13,26 +13,34 @@ public UserService() {
-    public void createUser(User user) {
+    public void registerUser(User user) {
         try {
+            if (userDAO.isUserExists(user.getEmail(), user.getUsername())) {
```
Убираем из всех классов stackTrace и заменяем на корректное логгирование

Willynick commented on starter/src/main/java/com/itacademy/courses/InsertApp.java:
```diff
@@ -45,39 +36,37 @@ public InsertApp() {
-        createTaskTag();
-        createProjectTask();
+        //createTag();
+        //createReportsForUser();
```
Приведи в порядок, удали ненужные комментарии и ненужные методы.

GeniusFromMinsk committed:
исправленное task_19

Внёс изменения в классы model(поставил отступы)
Создал универсальный метод фильтрации задач с помощью enum
Также вынес настройки подключения к базе данныз в файл .properties

GeniusFromMinsk committed:
commit
